pallets/xtransfer/src/xcm/xcm_helper.rs

```diff
@@ -291,11 +358,14 @@ pub mod xcm_helper {
 				&what,
 				&who,
 			);
-			if NativeChecker::is_native_asset(what) {
-				NativeAdapter::withdraw_asset(what, who)
+			let assets = if NativeChecker::is_native_asset(what) {
+				NativeAdapter::withdraw_asset(what, who).map_err(|e| return e)?
 			} else {
-				AssetsAdapter::withdraw_asset(what, who)
-			}
+				AssetsAdapter::withdraw_asset(what, who).map_err(|e| return e)?
+			};
+			WithdrawHandler::on_withdrawn(what.clone(), who.clone())
```

Comment:
```suggestion
			WithdrawHandler::on_withdrawn(what, who)
```

Reply:
We can not share reference to `on_withdraw` here, `Event` arguments with a reference will need us give a life time parameter, not sure which one is more effective

Reply:
Oh, got it. It's already a reference.